src/ip.py

```diff
@@ -125,11 +126,104 @@ def randomMAC():
     return ":".join(["%02x" % x for x in mac])
 
 
+# Decode ifconfig output
+def parse_ifconfig(res, address):
+    links = []
+    ifindex = 1
+
+    for r in res.split("\n"):
+        if re.match(r"^\w+:", r):
+            if ifindex > 1:
+                links.append(link)
+            (ifname, flags, mtu) = re.findall(r"^(\w+): flags=\d+<(.*)> mtu (\d+)", r)[0]
+            flags = flags.split(",")
+            link = {
+                "ifindex": ifindex,
+                "ifname": ifname,
+                "flags": flags,
+                "mtu": mtu
+            }
+            if "LOOPBACK" in flags:
+                link["link_type"] = "loopback"
+                link["address"] = "00:00:00:00:00:00"
+                link["broadcast"] = "00:00:00:00:00:00"
+            elif "POINTOPOINT" in flags:
+                link["link_type"] = "none"
+            else:
+                link["link_type"] = "unknown"
+            ifindex = ifindex + 1
+        else:
+            if re.match(r"^\s+ether ", r):
+                link["link_type"] = "ether"
+                link["address"] = re.findall(r"(\w\w:\w\w:\w\w:\w\w:\w\w:\w\w)", r)[0]
+                link["broadcast"] = "ff:ff:ff:ff:ff:ff"
+            elif address and re.match(r"^\s+inet ", r):
+                (local, netmask) = re.findall(r"inet (\d+\.\d+\.\d+\.\d+) netmask (0x[0-9a-f]+)", r)[0]
+                addr = {
+                    "family": "inet",
+                    "local": local,
+                    "prefixlen": netmask_to_length(netmask),
+                }
+                if re.match(r"^.*broadcast", r):
+                    addr["broadcast"] = re.findall(r"broadcast (\d+\.\d+\.\d+\.\d+)", r)[0]
+                link["addr_info"] = link.get("addr_info", []) + [addr]
+            elif address and re.match(r"^\s+inet6 ", r):
+                (local, prefixlen) = re.findall(r"inet6 ([0-9a-f:]*::[0-9a-f:]+)%*\w* prefixlen (\d+)", r)[0]
+                link["addr_info"] = link.get("addr_info", []) + [{
+                  "family": "inet6",
+                  "local": local,
+                  "prefixlen": int(prefixlen)
+                }]
+
+    links.append(link)
+
+    return links
+
+
+def link_addr_show(argv, af, address):
+    if len(argv) > 0 and argv[0] == "dev":
+        argv.pop(0)
+    if len(argv) > 0:
+        param = argv[0]
+    else:
+        param = "-a"
+
+    status, res = subprocess.getstatusoutput(
+        IFCONFIG + " " + param + " 2>/dev/null"
+    )
+    if status:  # unix status
+        if res == "":
+            perror(param + " not found")
+        else:
+            perror(res)
+        return False
+
+    links = parse_ifconfig(res, address)
+
+    if json_print:
```

Comment:
Just a small nit, but did you consider avoiding the `else` nesting and instead doing something like:

```
if json_print:
  return json_dump(...)
```

This applies to all the locations. I think this could be a bit more readable, while the control flow would be clear. And the return values could be still used to propagate exit code on the json path (I haven't reviewed that in detail).

Reply:
Sure, done